e2e-tests/src/test/button_game/mod.rs

```diff
@@ -55,6 +60,56 @@ pub fn marketplace(config: &Config) -> Result<()> {
     )?;
     marketplace.buy(&sign(&conn, player), None)?;
 
+    // Upgrade begins
+
+    // Performing code upgrade
+    let set_code_result = marketplace.set_code(
+        &sign(&conn, &authority),
+        config
+            .test_case_params
+            .marketplace_v2_code_hash
+            .as_ref()
+            .expect("New code's code_hash must be specified."),
+        None, // Alternative for performing "atomic" upgrade + migration (make sure that selector to migrate method is correct)
+              // Some("0x060d3f50".to_string())
+    );
+    info!("Trying to set code actual hash_code: {:?}", set_code_result);
+    assert!(set_code_result.is_ok());
+
+    // Change the metadata (keeping the old address)
+    marketplace = Arc::from(
+        MarketplaceInstance::new(
+            marketplace.contract().address().clone(),
+            &Some(
+                config
+                    .test_case_params
+                    .marketplace_v2_metadata
+                    .clone()
+                    .expect("New code's metadata path must be specified."),
+            ),
+        )
+        .expect("Adding existing contract should succeed."),
+    );
+
+    let migration_result = marketplace.migrate(&sign(&conn, &authority));
+    info!(
+        "Trying to perform migration after changing the metadata: {:?}",
+        migration_result
+    );
+    assert!(migration_result.is_ok());
+
+    // Check if migration was actually performed
+    //
+    // Will fail when:
+    // - Upgrade was not successful/not performed, or
+    // - Migration was not successful/not performed
```

Comment:
Suggestion: Same here, I think this comment should be a docstring for `migration_performed`.